### PR TITLE
Add support for outputting versions in JSON format

### PR DESCRIPTION
**Pull request by hughsie**
To use this, do: `fwupdmgr --version --json`

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-util.c [hunk 1]
```diff
@@ -3676,6 +3676,119 @@ fu_util_show_plugin_warnings(FuUtilPrivate *priv)
...
+	if (g_str_has_prefix(str, "CompileVersion(")) {
+		if (dependency_kind != NULL)
+			*dependency_kind = FU_UTIL_DEPENDENCY_KIND_COMPILE;
+		return g_strndup(str + 15, strsz - 16);
+	}
+	return g_strdup(str);
+}
+
+static gboolean
+fu_util_version_as_json(FuUtilPrivate *priv, GHashTableIter *iter, GError **error)
```
[the remark this answers is not in the document]
Comment (hughsie) [reply src/fu-util.c:3713]: > I think we want json support in both tools

Yes, I suppose so. Will respin.

File: src/fu-util.c [hunk 2]
```diff
@@ -3676,6 +3676,119 @@ fu_util_show_plugin_warnings(FuUtilPrivate *priv)
...
+		return "runtime";
+	if (dependency_kind == FU_UTIL_DEPENDENCY_KIND_COMPILE)
+		return "compile";
+	return NULL;
+}
+
+static gchar *
+fu_util_parse_project_dependency(const gchar *str, FuUtilDependencyKind *dependency_kind)
+{
+	gsize strsz = strlen(str);
```
Comment (superm1) [inline src/fu-util.c:3698]: I think you want a g_return_val_if_fail  check that dependency_kind is not a NULL pointer.

File: src/fu-util.c [hunk 3]
```diff
@@ -3676,6 +3676,119 @@ fu_util_show_plugin_warnings(FuUtilPrivate *priv)
...
+		g_string_append_printf(str,
+				       "%-10s%-30s%s\n",
+				       fu_util_dependency_kind_to_string(dependency_kind),
+				       project,
+				       value);
+	}
+
+	/* add this as a sanity check */
+	if (fwupd_client_get_host_bkc(priv->client) != NULL) {
+		g_string_append_printf(
```
Comment (superm1) [inline src/fu-util.c:3781]: isn't this missed in the json?

Comment (hughsie) [reply src/fu-util.c:3781]: Yup, I just noticed that too :) I actually think adding it to the report metadata is a good idea too, so we can understand "odd" upgrade paths.

File: src/fu-util.c (hunk 2 above)
Comment (hughsie) [reply src/fu-util.c:3698]: fixed

File: src/fu-util.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (hughsie) [reply src/fu-util.c:3713]: Done

**Review by superm1 — APPROVED** [review]

**Pull request merged**